### PR TITLE
Remove the `secondaryToolbarButton` CSS class

### PR DESCRIPTION
**Pull request by timvandermeij**
Secondary toolbar buttons are toolbar buttons with some extra rules, mainly to make them wider and have visible labels. However, this similarity is currently not clearly reflected in the implementation because the secondary toolbar buttons use a different CSS class, `secondaryToolbarButton`, compared to the other toolbar buttons that use the `toolbarButton` CSS class. This also causes some duplication in the rules and requires extra care to keep the common bits for the `secondaryToolbarButton` class in sync with the `toolbarButton` class.

Fortunately, now that we have a dedicated CSS scope for the secondary toolbar, we can simplify this by giving all secondary toolbar buttons the `toolbarButton` class and explicitly listing the required overrides in the `#secondaryToolbar` scope instead. Doing so removes most of the special-casing for secondary toolbar buttons while explicitly listing the differences in a single place for a better overview. It also lays the foundation for making all toolbar buttons respect the `browser.uidensity` Firefox preference later by reducing differences.

Extracts a part of #18385.

### CODE REVIEW

File: web/viewer.css [hunk 1]
```diff
@@ -1540,7 +1542,7 @@ dialog :link {
 }
 
 .visibleMediumView {
-  display: none;
+  display: none !important;
```
Comment (timvandermeij) [inline web/viewer.css:1545]: `!important` here and below is necessary because the rules inside a scope take precedence over rules outside of a scope, and within the secondary toolbar scope we now set the `display` property. Using `!important`  here should overall be safer to ensure that this always takes precedence for correct display in the viewer, but especially now that we're using CSS scoping more often it will also avoid issues for future patches.

File: web/viewer.css [hunk 2]
```diff
@@ -1264,6 +1223,49 @@ dialog :link {
...
+      opacity: var(--doorhanger-icon-opacity);
+      top: 5px;
+    }
+
+    &:is(:hover, :focus-visible) {
+      background-color: var(--doorhanger-hover-bg-color);
+      color: var(--doorhanger-hover-color);
+    }
+
+    & > span {
```
Comment (Snuffleupagus) [inline web/viewer.css:1251]: Is the `&` actually necessary, since I *believe* that it's implicit if written as follows?
```css
> span {
```

File: web/viewer.css [hunk 3]
```diff
@@ -1264,6 +1223,49 @@ dialog :link {
...
+      display: block;
+      height: auto;
+      padding-inline-end: 4px;
+      width: auto;
```
Comment (Snuffleupagus) [inline web/viewer.css:1255]: While it doesn't *seem* to break anything, I did notice when using the DOM inspector that these spans are now wider (previously the spans where only as big as the text, now they fill the entire button); was this change intentional?

File: web/viewer.css (hunk 2 above)
Comment (timvandermeij) [reply web/viewer.css]: You're right, it's not. I have removed it in the new commit.

File: web/viewer.css (hunk 3 above)
Comment (timvandermeij) [reply web/viewer.css]: Good catch; it was not expected, but it should indeed not break anything. Nevertheless I found a simpler solution using `display: unset` so we don't need the width/height overrides at all anymore and the span automatically fits the content again.

**Comment by Snuffleupagus** [comment]
In the `master` branch the `.toggled` class overrides the `:hover` effect, for the secondaryToolbar buttons, which this patch seem to "break"; I'm assuming this isn't intended?

**Comment by timvandermeij** [comment]
Good catch! The original rule only overrode the secondary toolbar button style for the non-toggled case but I forgot to take that along in the new rule. This is fixed in the new commit and hovering over the toolbar buttons is now the same as before again.

/botio-linux preview

**Review by Snuffleupagus — APPROVED** [review]
r=me, thank you.

File: web/viewer.css [hunk 4]
```diff
@@ -1264,6 +1223,47 @@ dialog :link {
...
+      color: var(--doorhanger-hover-color);
+    }
+
+    > span {
+      display: unset;
+      padding-inline-end: 4px;
+    }
+  }
+
+  a.toolbarButton {
```
Comment (Snuffleupagus) [inline web/viewer.css:1257]: Nit: Could this be shortened?
```suggestion
  &.a {
```

Comment (Snuffleupagus) [reply web/viewer.css:1257]: Or even moved into the `.toolbarButton` block above?

Comment (timvandermeij) [reply web/viewer.css:1257]: I would like that, but it'd basically require a way to, in nested CSS, detect on which element the class is being applied. I don't think there is way to do that (or at least I'm not aware of it).

Comment (Snuffleupagus) [reply web/viewer.css:1257]: It might be abusing notation a tiny bit, but the following does seem to work:
```diff
diff --git a/web/viewer.css b/web/viewer.css
index 7b1236e10..5da060f81 100644
--- a/web/viewer.css
+++ b/web/viewer.css
@@ -1252,15 +1252,15 @@ dialog :link {
       display: unset;
       padding-inline-end: 4px;
     }
-  }

-  a.toolbarButton {
-    padding-top: 5px;
-    text-decoration: none;
+    &:is(a) {
+      padding-top: 5px;
+      text-decoration: none;

-    &[href="#"] {
-      opacity: 0.5;
-      pointer-events: none;
+      &[href="#"] {
+        opacity: 0.5;
+        pointer-events: none;
+      }
     }
   }
```

Comment (timvandermeij) [reply web/viewer.css:1257]: Thanks; I'll give that a try and prepare a follow-up for this and another thing I noticed along the way.
